### PR TITLE
feat set data persistence for country

### PR DESCRIPTION
**Pull request by SallesCosta**
The country selection will be stored in localStorage under the key `clickplanet-country` to ensure data persistence.

### CODE REVIEW

File: src/app/viewer/useStorage.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
```
Comment (raphoester) [inline src/app/viewer/useStorage.ts:1]: this file should probably be called useCountryStorage.

Comment (raphoester) [reply src/app/viewer/useStorage.ts:1]: the function could be called `useCountryStorage` aswell ; useStorage is not precise enough

File: src/app/components/SelectWithSearch.tsx [hunk 2]
```diff
@@ -34,6 +38,9 @@ export default function SelectWithSearch(props: SelectWithSearchProps) {
         setSelected(value)
         setSearch("")
         props.onChange(value)
+
+        const toStorageValue = { name: value.name, code: value.code }
```
Comment (raphoester) [inline src/app/components/SelectWithSearch.tsx:42]: toStorageValue looks like a function name, why don't you just call it `storageValue` ?

File: src/app/components/SelectWithSearch.tsx [hunk 3]
```diff
@@ -13,9 +14,12 @@ type SelectWithSearchProps = {
 }
 
 export default function SelectWithSearch(props: SelectWithSearchProps) {
+
     const [search, setSearch] = useState("")
     const [selected, setSelected] = useState(props.selected)
 
+    const { handleSetCountry } = useStorage();
```
Comment (raphoester) [inline src/app/components/SelectWithSearch.tsx:21]: SelectWithSearch is a generic component that should not know anything about what it's selecting. Please abstract away the concept of country to keep the component as generic as possible.

**Review by raphoester — CHANGES_REQUESTED** [review]

File: src/app/viewer/useStorage.ts (hunk 1 above)
Comment (SallesCosta) [reply src/app/viewer/useStorage.ts]: @raphoester 
Name changed [here](src/app/viewer/useCountryStorage.ts).

File: src/app/components/SelectWithSearch.tsx (hunk 2 above)
Comment (SallesCosta) [reply src/app/components/SelectWithSearch.tsx]: Removed from this file to keep SelectWithSearch as generic as possible.

**Review by raphoester — APPROVED** [review]

**Pull request merged**